blog-posts/cloudfront-cache-efficiency.md

```diff
@@ -0,0 +1,154 @@
+body gets larger or the network speed of your customer decreases (a.e. on mobile plans).
+- Additional **traffic your customer has to pay** if the data is not free (a.e. on mobile plans).
+
+## Fixing the browsers cache behavior
+
+We spend hours and hours figuring out why the browser only caches the response for the `max-age` time, and then ignores 
+the cache for the rest of the `s-maxage` time. As it turns out, we were not the first ones how stumbled upon this problem:
+
+- [https://www.cdnplanet.com/blog/cloudfront-cachability-date-header/](https://www.cdnplanet.com/blog/cloudfront-cachability-date-header/)
+- [https://forums.aws.amazon.com/thread.jspa?messageID=807813](https://forums.aws.amazon.com/thread.jspa?messageID=807813)
+- [https://stackoverflow.com/a/61493383/3141881](https://stackoverflow.com/a/61493383/3141881)
+
+It turns out that the root of the problem is the `date` header returned by CloudFront. It doesn’t change on upcoming
+requests. As caching times in the `Cache-Control` header are relative to the response `date` header, only the first 
+request by the browser get's cached for the `max-age` time. 
+
+> CDNs usually don't cache the `date` header, it's only CloudFront that does it.
+
+There is no easy way to fix this within some CloudFront settings. Only recently it is possible to solve this problem at
+all: With a custom CloudFront Function. CloudFront Functions are small Javascript functions that are executed with every
+request. You can use them to do simple changes to the request or response object. Read more about the possibilities and
+limit in the official AWS docs
+here: [https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/functions-javascript-runtime-features.html](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/functions-javascript-runtime-features.html)
+
+We can use those functions to update the `date` header with every request.
+
+```js
+function handler(event) {
+  var response = event.response;
+
+  response.headers['date'] = {
+    value: new Date().toUTCString()
+  };
+  delete response.headers['age'];
+
+  return response;
+}
+```
+
+This code updates the `date` header with the current time. It also deletes the `age` header, as it turns out that
+CloudFront is also setting it. The [age header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Age) is used
+to tell how long the cached object is in the cache. If it is set, the browser will compare it against the `max-age` time
+and decide to not cache the object. Which is not what we want.
+
+After deploying the CloudFront function, let’s check the browser console:
+
+![Requests are cached by the browser](images/cloudfront-cache-efficiency/2-improved.png)
+
+What do we see here:
+
+1. The first request hits Lambda, so it takes around 700ms.
+2. The requests within 10 seconds are cached. After that, one request is done to CloudFront. This request takes around 30ms
```

Comment:
Is your expectation that the caching header from the source (us, 10 seconds) is forwarded to the CDN for the given resource? if yes, that is not clear to me.

I mean I understand the entire caching logic after reading your description for the given image. Every 10s the CDN is hit, every 60s the original source is hit (by the CDN). I also understand the date aspects and what the custom function does, but the knowledge for the cache calculation was not described in the "theory" before, which made this fix a surprise.